src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -92,15 +103,28 @@ def get(self, svc_id):
 
 class RgwRESTController(RESTController):
 
+    @Rgw.handle_error
     def proxy(self, method, path, params=None, json_response=True):
-        try:
-            instance = RgwClient.admin_instance()
-            result = instance.proxy(method, path, params, None)
-            if json_response and result != '':
-                result = json.loads(result.decode('utf-8'))
-            return result
-        except (DashboardException, RequestException) as e:
-            raise DashboardException(e, http_status_code=500, component='rgw')
+        instance = RgwClient.admin_instance()
+        result = instance.proxy(method, path, params, None)
+        if json_response:
+            result = json_str_to_dict(result)
+        return result
+
+
+@ApiController('/rgw/site', Scope.RGW)
+class RgwSite(RgwRESTController):
+
+    @Rgw.handle_error
+    def list(self, query=None):
+        if query == 'placement-targets':
+            result = RgwClient.get_placement_targets()
```

Comment:
Should we agree if we want Collections to be Arrays/Lists or Maps/Dicts? One big advantage for the Map approach is that you can address items by key, so no need for iterating. However, that forces you define what that unique key is for every Resource. I've always worked with Collections as `Array<Resource>` or `List[Resource]`.

Reply:
I'm open to discuss this. As an example, it seems that Github API returns dicts:
https://developer.github.com/v3/search/#example

Reply:
Well, it you see there, it returns an `Array` of `objects/dicts`, but no key-item collection. As, with default `json` package, we need to parse all and then process, we don't benefit too much from either. However, if we would replace that with some JSON parser allowing expressions (like [jsonpath](https://pypi.org/project/jsonpath-ng/)), we could improve this. It all depends on what the consumer side wants to do with this. Probably not a big issue right now, **except for aligning on returning the same kind of collection: `Array` or `Map`**

Reply:
This is an example of current response that I think it fits what you suggest:
```
{
        zonegroup: 'default',
        placement_targets: [
          {
            name: 'default-placement',
            data_pool: 'default.rgw.buckets.data'
          },
          {
            name: 'placement-target2',
            data_pool: 'placement-target2.rgw.buckets.data'
          }
        ]
      }
```

Reply:
Sorry, I made a mistake here. I assumed you were returning directly the placement-target dir, not nested.